src/utils/message_builder.rs

```diff
@@ -723,6 +727,190 @@ impl fmt::Display for MessageBuilder {
     }
 }
 
+
+/// Content modifier enum
+///
+/// Provides an enum of formatting modifiers for a string, for combination with
+/// string types and Content types.
+///
+/// # Examples
+///
+/// Create a new Content type which describes a bold-italic "text":
+///
+/// ```rust
+/// use serenity::utils::{Bold, Italic, Content};
+/// let content: Content = Bold + Italic + "text";
+/// ```
+pub enum ContentModifier {
+  Italic,
+  Bold,
+  Strikethrough,
+  Code,
+  Underline
+}
+
+/// Describes formatting on string content
+#[derive(Default, Clone)]
+pub struct Content {
+  italic: bool,
+  bold: bool,
+  strikethrough: bool,
+  pub inner: String,
+  code: bool,
+  underline: bool
+}
+
+impl Add<String> for Content {
+  type Output = Content;
+  fn add(self, rhs: String) -> Content {
+    let mut nc = self.clone();
+    nc.inner = nc.inner + &rhs;
+    nc
+  }
+}
+
+impl<'a> Add<&'a str> for Content {
+  type Output = Content;
+  fn add(self, rhs: &str) -> Content {
+    let mut nc = self.clone();
+    nc.inner = nc.inner + rhs;
+    nc
+  }
+}
+
+impl Add<String> for ContentModifier {
+  type Output = Content;
+  fn add(self, rhs: String) -> Content {
+    let mut nc = self.to_content();
+    nc.inner = nc.inner + &rhs;
+    nc
+  }
+}
+
+impl<'a> Add<&'a str> for ContentModifier {
+  type Output = Content;
+  fn add(self, rhs: &str) -> Content {
+    let mut nc = self.to_content();
+    nc.inner = nc.inner + rhs;
+    nc
+  }
+}
+
+impl Add<ContentModifier> for Content {
+  type Output = Content;
+  fn add(self, rhs: ContentModifier) -> Content {
+    let mut nc = self.clone();
+    nc.apply(&rhs);
+    nc
+  }
+}
+
+impl Add<ContentModifier> for ContentModifier {
+  type Output = Content;
+  fn add(self, rhs: ContentModifier) -> Content {
+    let mut nc = self.to_content();
+    nc.apply(&rhs);
+    nc
+  }
+}
+
+impl ContentModifier {
+  fn to_content(&self) -> Content {
+    let mut nc = Content::default();
+    nc.apply(self);
+    nc
+  }
+}
+
+impl Content {
+  pub fn apply(&mut self, modifier: &ContentModifier) {
+    match *modifier {
+      ContentModifier::Italic => {
+        self.italic = true;
+      },
+      ContentModifier::Bold => {
+        self.bold = true;
+      },
+      ContentModifier::Strikethrough => {
+        self.strikethrough = true;
+      },
+      ContentModifier::Code => {
+        self.code = true;
+      },
+      ContentModifier::Underline => {
+        self.underline = true;
+      }
+    }
+  }
+  pub fn to_string(&self) -> String {
+    let mut newstr = String::with_capacity(
+      self.inner.len()
+      + if self.bold {4} else {0}
+      + if self.italic {2} else {0}
+      + if self.strikethrough {4} else {0}
+      + if self.underline {4} else {0}
+      + if self.code {2} else {0}
+    );
+    if self.bold {
+      newstr.push_str("**");
+    }
+    if self.italic {
+      newstr.push_str("*");
```

Comment:
Since this is pushing only a char, [`String::push`](https://doc.rust-lang.org/std/string/struct.String.html#method.push) can be used.